src/lib.rs

```diff
@@ -131,26 +124,14 @@ impl StarSettings {
         ];
         StarSettings {
             reference_path: reference_path.to_string(),
-            multn: DEFAULT_MULTN,
             args: def_args,
         }
     }
 
-    /// Update the array of arguments when the arguments' values have changed
-    fn sync_args(&mut self) {
-        for i in 0..self.args.len() {
-            if self.args[i] == "--genomeDir" {
-                self.args[i + 1] = self.reference_path.clone();
-            } else if self.args[i] == "--outSAMmultNmax" {
-                self.args[i + 1] = self.multn.to_string();
-            }
-        }
-    }
-
-    /// Set the max number of multimapping reads
-    pub fn set_multn(&mut self, new_multn: isize) {
-        self.multn = new_multn;
-        self.sync_args();
+    /// Add an argument to pass to STAR.
+    pub fn arg(mut self, arg: &str) -> Self {
```

Comment:
This is really bikeshedding, but I would prefer something more akin to:
```
pub fn arg<N: AsRef<str>, V: AsRef<str>>(mut self, name: N, value: Option<V>) -> Self {
...
}
```
which would let us pass arguments and their values (if they exist) separately. But again, this is probably fine for most things.

Ultimately it would be better to figure out which bits we want to fiddle from the rust side and develop a proper API (like cc::Build) to set the appropriate arguments.

Reply:
There are arguments for both styles, but I prefer to mirror the same interface as the standard library function [`std::process::Command::arg`](https://doc.rust-lang.org/std/process/struct.Command.html#method.arg)

Reply:
Happy to take the `AsRef<str> + Display` suggestion if you prefer that over `&str` though.

Reply:
Eh, this is fine. And that's a good argument. :+1:

Reply:
Hmm, actually, it may be better to store args as a hashmap and override them? Not sure how star's argument-parsing ability handles things being overridden later.

Reply:
In fact, the way you have it, it may be better to move all the default arguments we use out of here and into CR.

Reply:
> Hmm, actually, it may be better to store args as a hashmap and override them? Not sure how star's argument-parsing ability handles things being overridden later.

I tested this, and it gives an error if the same argument is specified more than once.

> In fact, the way you have it, it may be better to move all the default arguments we use out of here and into CR.

I don't disagree, but this late before branching, I'm inclined to leave it as is, and do that in a future PR if we want to tidy it up.